### DISCUSSION TITLE
How to avoid quotes for Integers, Boolean values in JSON output

### DISCUSSION THREAD

**Discussion by ajit4sbi** [mikefarah/yq#1478] [Q&A] [answered]
Please provide command tricks to get below expected JSON output without quotes to Integers, Boolean values. 

****** XML INPUT ******

`  <outputImageProperties>
    <printerDPI>200</printerDPI>
    <encodingFormat>pdf</encodingFormat>
    <imageOptions>
      <typeCode>label</typeCode>
      <templateName>ECOM26_84_A4_001</templateName>
    </imageOptions>
    <imageOptions>
      <typeCode>waybillDoc</typeCode>
      <templateName>ARCH_8X4_A4_002</templateName>
      <isRequested>true</isRequested>
      <hideAccountNumber>true</hideAccountNumber>
      <numberOfCopies>1</numberOfCopies>
    </imageOptions>
    <imageOptions>
      <typeCode>invoice</typeCode>
      <templateName>COMMERCIAL_INVOICE_P_10</templateName>
      <isRequested>true</isRequested>
    </imageOptions>
`
 

****** Expected JSON Output ******

"outputImageProperties": {
        "printerDPI": 200,
        "encodingFormat": "pdf",
        "imageOptions": [
            {
                "typeCode": "label",
                "templateName": "ECOM26_84_A4_001"
            },
            {
                "typeCode": "waybillDoc",
                "templateName": "ARCH_8X4_A4_002",
                "isRequested": true,
                "hideAccountNumber": true,
                "numberOfCopies": 1
            },
            {
                "typeCode": "invoice",
                "templateName": "COMMERCIAL_INVOICE_P_10",
                "isRequested": true
            }
        ],

**Comment by mikefarah** [comment, marked as answer]
The XML parser assumes everything is a string - you can do this to attempt to parse all the scalar values into their types:
```
yq -p=xml ' (.. | select(tag == "!!str")) |= from_yaml' sample.xml
```

This selects all the string values and runs them through 'from_yaml'. That function will convert to numbers/floats/booleans if it can, otherwise it will leave the string value as is.

**Reply by mikefarah** [reply]
See https://mikefarah.gitbook.io/yq/usage/xml#parse-xml-number
